frontend/public/components/utils/horizontal-nav.tsx

```diff
@@ -258,9 +260,13 @@ export const HorizontalNav = React.memo((props: HorizontalNavProps) => {
   );
 
   const objReference = props.obj?.data ? referenceFor(props.obj.data) : '';
-  const [dynamicNavTabExtensions, navTabExtentionsResolved] = useResolvedExtensions<
-    DynamicHorizontalNavTab
-  >(DynamicIsHorizontalNavTab);
+  const contextId = props.contextId;
+  const [dynamicResourceNavTabExtensions, navTabResourceExtentionsResolved] = useResolvedExtensions<
+    DynamicResourceNavTab
+  >(DynamicIsResourceNavTab);
+  const [dynamicTabExtensions, navExtentionsResolved] = useResolvedExtensions<DynamicNavTab>(
+    DynamicIsNavTab,
+  );
```

Comment:
Looks like we've been doing it wrong this whole time.
It's incorrect to resolve the extensions here. We don't want to resolve code references that aren't relevant. Furthermore we only want to resolve the code reference when the tab is actually loaded.
use `useExtensions` instead.